### ISSUE TITLE
[GNS3 3.0.0] All custom appliances disappeared + Unable to reimport

### ISSUE THREAD

**Issue by ajbxfinest24** [GNS3/gns3-gui#3678]
**Describe the bug**
I took a timeshift snapshot of the operating system before I upgraded GNS3 from 2.2.52 to 3.0.0. The install was successful, but upon launching the new application all of my custom appliances disappeared. I was able to install ipterm, but when I attempted to reimport Cisco, Arista, F5 or Fortigate appliances I get the error below

****** Exception detected, traceback information saved in /home/xxx/.config/GNS3/3.0/exceptions.log ******

PLEASE REPORT ON https://www.gns3.com

TypeError: invalid result from ApplianceWizard.validateCurrentPage(), a 'bool' is expected not 'NoneType'

I am able to launch my old projects, but when I attempted to connect to my Arista appliance a booting loop occurs. Photos are attached. 
![Screenshot from 2024-12-21 20-20-04](https://github.com/user-attachments/assets/dd31682d-8d66-48ff-83b2-a7168f0fbfb2)
![Screenshot from 2024-12-21 20-19-29](https://github.com/user-attachments/assets/d3fe7b14-50aa-4268-ba68-ddab1c5e8972)

**GNS3 version and operating system (please complete the following information):**
 - OS: Linux Mint
 - GNS3 version 3.0.0

**Comment by grossmj** [comment]
What Cisco appliance have you tried to install?

**Comment by smart2128** [comment]
I can confirm the same behavior on Arch Linux for every appliance I try to install (e.g. VyOS, FRR, Cisco IOSvLx, RockyLinux, Kali)
I'm not sure if this might be related to the new Python 3.13.

**Comment by ajbxfinest24** [comment]
Just to confirm, I am able to open some of my previous projects. It loads the appliances configured for the project even the appliances not listed in the all devices column which is strange. Some of my projects that contained the network_automation docker image gives me an error message and I can't open it anymore

 "Could not pull the 'gns3/network_automation:latest' image from Docker Hub, please check your Internet connection (original error: Docker has returned an error: 404 pull access denied for gns3/network_automation, repository does not exist or may require 'docker login': denied: requested access to the resource is denied)"

So it seems to be hit or miss when it comes to specific projects. I don't mind starting over with re-importing all appliances, but right now that portion doesn't work and I can't use this software. I will look out for your recommendations.

**Comment by smart2128** [comment]
> Just to confirm, I am able to open some of my previous projects. It loads the appliances configured for the project even the appliances not listed in the all devices column which is strange. Some of my projects that contained the network_automation docker image gives me an error message and I can't open it anymore

Templates are not migrated automatically, you have to re-create them manually, as already answered by @grossmj in my closed issue [#3667](https://github.com/GNS3/gns3-gui/issues/3677) 

> "Could not pull the 'gns3/network_automation:latest' image from Docker Hub, please check your Internet connection (original error: Docker has returned an error: 404 pull access denied for gns3/network_automation, repository does not exist or may require 'docker login': denied: requested access to the resource is denied)"

There is no gns3/network_automation in docker hub, that's probably why it returns that error.
If I add the network_automation container from the wizard, it pulls a different image (adosztal/network_automation).

**Comment by ajbxfinest24** [comment]
> > Just to confirm, I am able to open some of my previous projects. It loads the appliances configured for the project even the appliances not listed in the all devices column which is strange. Some of my projects that contained the network_automation docker image gives me an error message and I can't open it anymore
> 
> Templates are not migrated automatically, you have to re-create them manually, as already answered by @grossmj in my closed issue [#3667](https://github.com/GNS3/gns3-gui/issues/3677)
> 
> > "Could not pull the 'gns3/network_automation:latest' image from Docker Hub, please check your Internet connection (original error: Docker has returned an error: 404 pull access denied for gns3/network_automation, repository does not exist or may require 'docker login': denied: requested access to the resource is denied)"
> 
> There is no gns3/network_automation in docker hub, that's probably why it returns that error. If I add the network_automation container from the wizard, it pulls a different image (adosztal/network_automation).

Yes, i wasn't aware from the e-mail notification from github that the templates and appliances would not be migrated from 2.2 -> 3.0. If i received that in the notification like it's displayed on the blog currently I would've held off until there was a proper format with creating templates and importing a device. I tried to create a template manually from scratch and my Arista appliance doesn't boot up. Maybe I will continue to work on this at some point during the holiday break, but I attempted to restore my snapshot on my linux machine to a state with 2.2.52 installed, but because the project was opened using 3.0 It says I can't open it on 2.2.52, because it was created on a newer version of gns3 which that is not the case. So I have to use 3.0 after all but I can't get it to work.

**Comment by grossmj** [comment]
Sorry about these issues. I am working on a fix and will probably release v3.0.1 soon.

**Comment by grossmj** [comment]
@ajbxfinest24 I believe you use Linux Mint right? How have you installed version 3.0? Thanks 👍

**Comment by ajbxfinest24** [comment]
> @ajbxfinest24 I believe you use Linux Mint right? How have you installed version 3.0? Thanks 👍

@grossmj Yes, I use Linux Mint 22. I installed it from the ppa that you provided to me in the GNS3 3.0 RC2 response to my question 

sudo add-apt-repository ppa:gns3/ppa-v3
sudo apt install gns3-gui

**Comment by grossmj** [comment]
Unfortunately I installed a fresh Linux Mint 22 and couldn't reproduce the issue.

When do you get this error?

```
TypeError: invalid result from ApplianceWizard.validateCurrentPage(), a 'bool' is expected not 'NoneType'
```

**Comment by smart2128** [comment]
![Schermata_20241226_103209](https://github.com/user-attachments/assets/b98ecc17-60b5-47d8-bc3a-56e03bf89443)
In my case, but I believe it's the same for @ajbxfinest24, it happens every time I try to install a Qemu appliance from the GNS3 server using the appliance wizard.
No issues with Docker's appliances or manually created ones.

**Comment by grossmj** [comment]
I finally found the problem. Thanks everyone for your help!

This will be fixed in v3.0.1 to be released in the next few days.
